runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -1316,6 +1328,23 @@ private void onMergedInitialFlush(
                 doFetchPartitionsIfNecessary(traceId);
                 doProducePartitionsIfNecessary(traceId);
             }
+            else
```

Comment:
Line `1305` above is where we would switch out the filters, right?

Reply:
Nope, as that only happens if the capabilities have changed. We need to switch the streams in all cases when we have fetch capabilities.